pkg/controllers/mcs/endpointslice_controller.go

```diff
@@ -55,24 +56,17 @@ func (c *EndpointSliceController) Reconcile(ctx context.Context, req controllerr
 	work := &workv1alpha1.Work{}
 	if err := c.Client.Get(ctx, req.NamespacedName, work); err != nil {
 		if apierrors.IsNotFound(err) {
-			// Cleanup derived EndpointSlices after work has been removed.
-			err = helper.DeleteEndpointSlice(c.Client, labels.Set{
-				workv1alpha1.WorkNamespaceLabel: req.Namespace,
-				workv1alpha1.WorkNameLabel:      req.Name,
-			})
-			if err == nil {
-				return controllerruntime.Result{}, nil
-			}
```

Comment:
This logic can remain, refer to https://github.com/karmada-io/karmada/pull/4737#discussion_r1537004832

Reply:
yes, make sense.

Reply:
I think there will be a problem here, which is that the label `work.karmada.io/permanent-id` cannot be obtained.

Reply:
Yes, the problem seems unsolvable. Any ideas?

Reply:
If we want to obtain the label, we must use `util.AsyncWorker` to reconcile, but such a change will be huge. Is it worth it?

Reply:
Maybe we can do that in the release-note and in the upgrade documentation. Ask some ideas from @RainbowMango @chaunceyjiang

Reply:
I don't understand why removing this(cleanup derived EndpointSlices), don't need to do that anymore?

Reply:
As we discussed above, we now need to clean up derived EndpointSlices based on `work.karmada.io/permanent-id`, but it cannot be obtained here.

Reply:
Can we find the derived EndpointSlices by listing all of them? It's a tradeoff in addition to upgrading instruction mentioned by @XiShanYongYe-Chang on above.

Reply:
Of course, it's a good idea.